python/sglang/srt/layers/moe/ep_moe/token_dispatcher.py

```diff
@@ -248,8 +251,8 @@ def dispatch_normal(
             topk_idx,
             num_experts,
             previous_event=previous_event,
-            async_finish=False,
-            allocate_on_comm_stream=False,
+            async_finish=self.async_finish,
+            allocate_on_comm_stream=(previous_event is not None) and self.async_finish,
```

Comment:
Only the first condition is useful. If `previous_event` is not None, self.async_finish is always true.